dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -418,6 +434,135 @@ public void Test_Secret_Destroy_Method() {
+            throws IOException, DotDataException, DotSecurityException {
+        Logger.info(AppsAPIImplTest.class, ()->"Evaluating  "+testCase.toString());
+        final AppDescriptorDataGen descriptorDataGen = new AppDescriptorDataGen();
+        descriptorDataGen.withName(testCase.name).withKey(testCase.key).withDescription(testCase.description).withIconUrl(testCase.iconUrl);
+        for (final Map.Entry<String, ParamDescriptor> entry : testCase.params.entrySet()) {
+            descriptorDataGen.param(entry.getKey(), entry.getValue());
+        }
+        try (final InputStream inputStream = descriptorDataGen.nextPersistedDescriptor()) {
+            final AppsAPI api = APILocator.getAppsAPI();
+            final User admin = TestUserUtils.getAdminUser();
+            return api.createAppDescriptor(inputStream, admin);
+        }
+    }
+
+    /**
+     * Test AppsAPI#createAppDescriptor input validation.
+     * Given scenario: Passing Params that expect to break the validation rules
+     * Expected Result: DotDataValidationException is raised.
+     */
+    @Test(expected = DotDataValidationException.class)
+    @UseDataProvider("getExpectedExceptionTestCases")
+    public void Test_App_Descriptor_Validation_Expect_Validation_Exceptions(final AppTestCase testCase)
+            throws IOException, DotDataException, DotSecurityException {
+        assertNotNull(evaluateAppTestCase(testCase));
+    }
+
+    @DataProvider
+    public static Object[] getExpectedExceptionTestCases() throws Exception {
+        final Map<String, ParamDescriptor> emptyParams = ImmutableMap.of();
+        return new Object[]{
+                //The following test that the general required fields are mandatory
+                new AppTestCase("", "","", "", emptyParams),
+                new AppTestCase("any-key", "","", "",  emptyParams),
+                new AppTestCase("any-key", "any-name","", "", emptyParams),
+                new AppTestCase("any-key", "any-name","desc", "", emptyParams),
+                new AppTestCase("any-key", "any-name","desc", "icon", emptyParams),
+                //Name too large.
+                new AppTestCase(RandomStringUtils.randomAlphanumeric(AppsAPIImpl.DESCRIPTOR_KEY_MAX_LENGTH + 1), "any-name","desc", "icon", emptyParams),
+                //Key-too large
+                new AppTestCase("any-key", RandomStringUtils.randomAlphanumeric(AppsAPIImpl.DESCRIPTOR_NAME_MAX_LENGTH + 1),"desc", "icon", emptyParams),
+                //The following test paramDefinition
+                //Null type  is not allowed.
+                new AppTestCase("any-key", "any-name", "desc", "icon", ImmutableSortedMap.of(
+                  "p1", newParam("", true, null, "", "", true)
+                )),
+                //Hidden bool param is not allowed.
+                new AppTestCase("any-key", "any-name", "desc", "icon", ImmutableSortedMap.of(
+                  "p1", newParam("", true, Type.BOOL, "label", "hint", true)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "label" appears 8 times in this file; the first occurrence is on line 485](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)